config/methods/common_methods.config

```diff
@@ -391,6 +391,63 @@ methods {
         process.containerOptions  = {-> (task.containsKey('cpus')) ? "--cpu-shares ${default_cpu_shares} --cpus ${task.cpus}" : "--cpu-shares ${default_cpu_shares}"}
     }
 
+    /**
+     * Configure all processes to save their command files in the output
+     * directory.
+     *
+     * This add a custom process directive that, when used as the afterScript,
+     * will copy all of the process's .command.* files into the output
+     * directory.
+
+     * Processes can customize the output directory by setting
+     * `process.ext.log_dir` and `process.ext.log_dir_suffix`. Both may be
+     * closures.
```

Comment:
This is good; many of our processes end up adding something useful (ex. sample ID, interval ID, task index) to the directory name along with the process name to differentiate the process logs from the same process running multiple times. We'll likely want to try it out with a pipeline or process that does this

Reply:
Yeah - here's my toy example from the demo pipeline that creates output folders `compute_sum-even` and `compute_sum-odd`. We definitely should test it with more complex cases.

```groovy
process compute_sum {
    label 'process_low'
    container 'ubuntu:22.10'

    publishDir path: "${params.output_dir_base}/output",
        mode: "copy",
        pattern: "*.txt"

    afterScript {
        lines = [
            "echo 'Before the common'",
            task.ext.commonAfterScript ?: "",
            "echo 'After the common'"
        ]
        return lines.join("\n")
    }

    ext log_dir_suffix: { "-${type}" }

    input:
    tuple val(type), path(integers)

    ...
```